packages/playwright-core/src/server/chromium/crNetworkManager.ts

```diff
@@ -153,19 +153,34 @@ export class CRNetworkManager {
 
   _onAuthRequired(event: Protocol.Fetch.authRequiredPayload) {
     let response: 'Default' | 'CancelAuth' | 'ProvideCredentials' = 'Default';
+    const shouldProvideCredentials = this._shouldProvideCredentials(event.request.url);
     if (this._attemptedAuthentications.has(event.requestId)) {
       response = 'CancelAuth';
-    } else if (this._credentials) {
+    } else if (shouldProvideCredentials) {
       response = 'ProvideCredentials';
       this._attemptedAuthentications.add(event.requestId);
     }
-    const { username, password } = this._credentials || { username: undefined, password: undefined };
+    const { username, password } =  shouldProvideCredentials && this._credentials ? this._credentials : { username: undefined, password: undefined };
     this._client._sendMayFail('Fetch.continueWithAuth', {
       requestId: event.requestId,
       authChallengeResponse: { response, username, password },
     });
   }
 
+  _shouldProvideCredentials(url: string): boolean {
+    if (this._credentials){
```

Comment:
In JavaScript, we would say

```js
_shouldProvideCredentials(url: string): boolean {
  if (!this._credentials)
    return false;
  return !this._credentrials.hostname || new URL(url).hostname.toLowerCase() === this._credentials.hostname.toLowerCase();
}


return this._credentials;
```

Reply:
I agree, this is more condensed, I just fixed it.